src/release/rollups.md

```diff
@@ -23,67 +23,99 @@ queue has been merged.
+   provides to bisect the failure to a specific PR and
+   `@bors r-` the PR. If the PR is running, you need to do `@bors r- retry`. Otherwise,
    your rollup succeeded. If it did, proceed to the next rollup (every now and
-   then let `rollup=never` and toolstate PRs progress).
-4. Recreate the rollup without the offending PR starting again from **1.**. There's a link in the rollup PR's body to automatically prefill the rollup UI with the existing PRs (minus any PRs that have been `r-`d)
+   then let `rollup=never/iffy` and toolstate PRs progress).
+4. Recreate the rollup without the offending PR starting again from **1.**. 
+   There's a link in the rollup PR's body to automatically prefill the rollup UI 
+   with the existing PRs (minus any PRs that have been `r-`d)
+   Try avoiding adding any additional PR to the current "batch" as this
+   unnecessarily increases the chance of test failures.
+   Rollups should not overlap, if a PR is already contained in a rollup that is not closed,
+   it should not be added to another different rollup at the same time.
+   If a rollup fails and you are not sure which PR caused the problem, 
+   you may bisect the rollup and split it up into two rollups until the offending PR becomes clear.
 
 ## Selecting Pull Requests
 
-The queue is sorted by rollup status. In general, a good rollup includes one or two `iffy` PRs (if available), a bunch of `maybe` (unmarked) PRs, and a large pile of `always` PRs. A rollup should never include `rollup=never` PRs.
+The queue is sorted by rollup status. In general, a good rollup contains a bunch of `maybe` (unmarked) PRs, and a large pile of `always` PRs. You can include one or two `iffy` PRs if you are confident that they will pass. 
+If you have a lot of time, you can also make a rollup just from `iffy` PRs (if there are enough of them) and weed out the failures one by one.
+A rollup should never include `rollup=never` PRs.
+
+The actual absolute size of the rollup can depend based on experience and current length of the queue.
+People new to making rollups might start with including 1 `iffy`, 2 `maybe`s, and 4 `always`s. Usually 6-8 PRs per rollup is a good compromise.
+There is rarely a need to roll up more than 10 PRs at once (unless there are >30 PRs waiting the queue), keep in mind that we also try to minimize regressions per merge.
 
-The actual absolute size of the rollup can depend based on experience, people new to making rollups might start with including 1 `iffy`, 4 `maybe`s, and 5 `always`s, but more experienced people might even make a rollup of 1-2 `iffy`s, 8 `maybe`s, and 10 `always`s! Massive rollups are rarely needed, but as your intuition grows you'll get better at judging risk when including PRs in a rollup.
+Don't try to make mega-rollups (15-20 PRs that merge half or more of the entire queue all at once) to keep the number of perf or bug regressions per merge as low as possible and keep potential regressions [bisectable].
```

Comment:
```suggestion
Limit the size of rollups, even if the queue is backed up -- large rollups run the risk of failing or merge conflicts, and smaller rollups keep potential regressions [bisectable]. On average, rollups are <N> PRs large, often varying from <N - M> to <N + M> depending on the number of `rollup=always` PRs that can be included. 
```

Reply:
choose some value for `N` and `M`.

Reply:
note that thanks to unrolled builds, bisection can be done within a roll-up and cargo bisect rustc